### ISSUE TITLE
@solid-auth/oauth2 node.js crypto dependency

### ISSUE THREAD

**Issue by MaryHal** [OrJDev/solid-auth#23]
First off, thanks for these packages, they were pretty easy to integrate into one of my projects!

One issue I had is that my deployment target is Cloudflare Workers, so when I tried deploying there, I was getting this error:

```
✘ [ERROR] Could not resolve "crypto"

    dist/server.js:2:21:
      2 │ import crypto$1 from 'crypto';
        ╵                      ~~~~~~~~

  The package "crypto" wasn't found on the file system but is built into node.
  Add "node_compat = true" to your wrangler.toml file to enable Node compatibility.


✘ [ERROR] Build failed with 1 error:

  dist/server.js:2:21: ERROR: Could not resolve "crypto"
```

`node_compat` doesn't implement enough to Just Work. The problem is that @solid-auth/oauth2 uses the uuid package which uses the node.js crypto module. Is it possible to optionally generate uuids with WebCrypto `crypto.randomUUID` instead (in `generateState`)?

Whipped up a [repro repo](https://github.com/MaryHal/solid-auth-repro) using create-jd-app. `pnpm install`, `pnpm run publish` should do the trick.

**Comment by OrJDev** [comment]
Yes i will dig more into it and will let you know

**Comment by OrJDev** [comment]
im no Cloudflare expert but wouldn't ```node_compat = true``` in ```wrangler.toml``` solve it?

**Comment by MaryHal** [comment]
While that does fix the build issue, the crypto implementation doesn't appear complete, so you end up seeing this when it's actually running:
![Bx0JcRpi0r](https://user-images.githubusercontent.com/1450355/206272395-9d0ffb24-a5e1-43bf-b63c-720ac8c3ca2d.png)

**Comment by MaryHal** [comment]
Not sure what I was thinking, this shouldn't be running in the browser, I'll look at it a bit more

**Comment by schweden1997** [comment]
The issues is that the uuid package uses the node crypto api. As far as I know CF only supports the Web Crypto API. We would need to change to a UUID package based on the standard Web api or use the standard library itself.

https://github.com/OrJDev/solid-auth/blob/4c5f8b88cfd9d1c106eb4106624f18783498e454/packages/oauth2/src/index.ts#L293-L295

returning `crypto.randomUUID()` generates a UUID v4. This is available in Node >= v14.17.0

I wasn't able to test this since I don't know how to properly setup a solid-start app on CF.

**Comment by MaryHal** [comment]
That was my understanding too, though I don't think I expressed it clearly. I've been patching the library to use `crypto.randomUUID()` and it's been working fine.
